pallets/oracle-data-collection/src/lib.rs

```diff
@@ -312,12 +357,31 @@ pub mod pallet {
 				})
 				.collect::<Result<Vec<_>, _>>()?;
 
-			T::AggregationProvider::aggregate(fed_values)
-				.ok_or(Error::<T>::KeyNotInCollection.into())
+			if fed_values.len() < (min_feeders as usize) {
+				Err(Error::<T>::NotEnoughFeeders)?
+			}
+
+			let updated_fed_values = fed_values
+				.into_iter()
+				.filter(|(_, timestamp)| {
+					Self::ensure_valid_timestamp(collection_id, *timestamp).is_ok()
+				})
+				.collect::<Vec<_>>();
+
+			if updated_fed_values.len() < (min_feeders as usize) {
+				Err(Error::<T>::OracleValueOutdated)?
+			}
+
+			let (value, timestamp) = T::AggregationProvider::aggregate(updated_fed_values)
+				.ok_or(Error::<T>::KeyNotInCollection)?;
+
+			Ok((value, timestamp))
 		}
 
-		fn collection(collection_id: &T::CollectionId) -> Self::Collection {
-			CachedCollection(Collection::<T>::get(collection_id))
+		fn collection(collection_id: &T::CollectionId) -> Result<Self::Collection, DispatchError> {
+			let collection = Collection::<T>::get(collection_id);
+			Self::ensure_valid_timestamp(collection_id, collection.older_timestamp)?; //TODO
```

Comment:
What is left TOdo here?

Reply:
Something I should have been removed 😅. It's already fixed.